services/src/messaging/src/main/java/org/openkilda/messaging/info/flow/FlowPathResponse.java

```diff
@@ -47,71 +46,15 @@
      * The response payload.
      */
     @JsonProperty(Utils.PAYLOAD)
-    protected PathInfoData payload;
+    protected ImmutablePair<PathInfoData, PathInfoData> payload;
```

Comment:
ImmutablePair is an ancient horror. We have a BidirectionalFlow as a replacement for this horror. It would be perfect to not use ImmutablePair in any newly created code.

Reply:
Ok, but the name of storm path action and classes (eg StreamType.PATH, FlowPathResponse etc) will no longer correspond with its returning payload (entire BidirectionalFlow). Should I rename all this artifacts?

Reply:
My fail... don't noticed that it return only part of flow.

But, why not return whole flow, extract only path in NB and send response. In this case we can drop this FlowPathResponse object and use... some existing object that incapsulate flow... Quick overview show that we don't have an object that encapsulate flow into InfoData.

Ok, create new InfoData successor, insert there BidirectionalFlow and use it for this conversation. Later it should be used in "conversation" returning flow object.

Reply:
So it's actually renaming PATH conversation to, for example, READ_BIDIRECTIONAL